File: app-wear/src/main/java/de/tu_darmstadt/seemoo/nfcgate/wear/MainActivity.java
package de.tu_darmstadt.seemoo.nfcgate.wear;

import android.os.Bundle;
import android.widget.Button;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

import de.tu_darmstadt.seemoo.nfcgate.wear.model.CaptureState;

public class MainActivity extends AppCompatActivity {
    private CaptureController controller;
    private TextView statusView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        controller = new CaptureController(this);
        statusView = findViewById(R.id.capture_status);
        Button startButton = findViewById(R.id.start_button);
        Button stopButton = findViewById(R.id.stop_button);
        Button pauseButton = findViewById(R.id.pause_button);
        startButton.setOnClickListener(v -> {
            controller.start();
            render(controller.getState());
        });
        stopButton.setOnClickListener(v -> {
            controller.stop();
            render(controller.getState());
        });
        pauseButton.setOnClickListener(v -> {
            controller.pause();
            render(controller.getState());
        });
        render(controller.getState());
    }

    private void render(CaptureState state) {
        String status = state.isActive() ? (state.isPaused() ? getString(R.string.status_paused) : getString(R.string.status_active)) : getString(R.string.status_idle);
        statusView.setText(getString(R.string.capture_status_template, status, state.getSessionCount(), state.getBytesCaptured()));
    }
}

File: android/hce-app/app/src/main/java/com/nfcgate/hce/ui/MainActivity.kt
package com.nfcgate.hce.ui

import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.LinearLayout
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.nfcgate.hce.relay.RelayManager

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        val root = LinearLayout(this).apply {
            orientation = LinearLayout.VERTICAL
            setPadding(48, 48, 48, 48)
        }

        val urlInput = EditText(this).apply {
            hint = "ws://server:8080/ws/relay"
            setText("ws://10.0.2.2:8080/ws/relay")
        }
        val tokenInput = EditText(this).apply {
            hint = "Relay token"
        }
        val status = TextView(this).apply {
            text = "Not connected"
        }
        val startButton = Button(this).apply {
            text = "Start Relay"
            setOnClickListener {
                val server = urlInput.text.toString().trim()
                val token = tokenInput.text.toString().trim()
                if (server.isEmpty() || token.isEmpty()) {
                    Toast.makeText(context, "URL and token required", Toast.LENGTH_SHORT).show()
                    return@setOnClickListener
                }

                val client = RelayManager.startRelay(token, server, "hce")
                client.setListener(object : com.nfcgate.hce.relay.RelayClient.RelayListener {
                    override fun onConnected(sessionId: String) {
                        runOnUiThread { status.text = "Connected: $sessionId" }
                    }

                    override fun onDisconnected(reason: String) {
                        runOnUiThread { status.text = "Disconnected: $reason" }
                    }
                })
            }
        }

        root.addView(urlInput)
        root.addView(tokenInput)
        root.addView(startButton)
        root.addView(status)

        setContentView(root)
    }

    override fun onDestroy() {
        super.onDestroy()
        RelayManager.stopRelay("activity_destroy")
    }
}
